components/support/migration/src/main/java/mozilla/components/support/migration/FennecMigrator.kt

```diff
@@ -368,17 +371,41 @@ class FennecMigrator private constructor(
         return getMigrationsToRun().isNotEmpty()
     }
 
+    private fun isFennecInstallation(): Boolean {
+        // Not implemented yet.
+        // https://github.com/mozilla-mobile/android-components/issues/5115
+        return true
+    }
+
     /**
-     * Starts the provided [AbstractMigrationService] implementation if there are migrations to be
-     * run for this installation.
+     * If a migration is needed then invoking this method will update the [MigrationStore] and launch
+     * the provided [AbstractMigrationService] implementation.
      */
-    fun <T> startMigrationServiceIfNeeded(service: Class<T>) where T : AbstractMigrationService {
-        if (hasMigrationsToRun()) {
-            logger.debug("Has migrations to run. Starting service.")
-            context.startService(Intent(context, service))
-        } else {
-            logger.debug("No migrations to run. Not starting service.")
+    fun <T> startMigrationIfNeeded(
+        store: MigrationStore,
+        service: Class<T>
+    ) where T : AbstractMigrationService {
+        if (!isFennecInstallation()) {
+            // This installation seems to never have been Fennec, so we do not need
+            // to migrate anything.
+            logger.debug("This is not a Fennec installation. No migration needed.")
+            return
+        }
+
+        if (!hasMigrationsToRun()) {
+            // There are no migrations to run for this installation. This likely means that we are
+            // migrated already and there are no updated migrations to run.
+            logger.debug("This is a Fennec installation. But there are no migrations to run.")
+            return
         }
+
+        logger.debug("Migration is needed. Updating state and starting service.")
+
+        runBlocking {
```

Comment:
Why block on this?

Reply:
We want this state to be available immediately and do not want to be reactive here (e.g. launch into the browser and then into the migration once the state has changed).